xbmc/platform/posix/storage/discs/DiscDriveHandlerPosix.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DiscDriveHandlerPosix.h"
+
+#include "storage/cdioSupport.h"
+#include "utils/log.h"
+
+#include <memory>
+
+namespace
+{
+constexpr int MaxOpenRetries = 3;
+}
+using namespace MEDIA_DETECT;
+
+std::shared_ptr<IDiscDriveHandler> IDiscDriveHandler::CreateInstance()
+{
+  return std::make_shared<CDiscDriveHandlerPosix>();
+}
+
+DriveState CDiscDriveHandlerPosix::GetDriveState(const std::string& devicePath)
+{
+  DriveState driveStatus = DriveState::DRIVE_NOT_READY;
+  const std::shared_ptr<CLibcdio> libCdio = CLibcdio::GetInstance();
+  if (!libCdio)
+  {
+    CLog::LogF(LOGERROR, "Failed to obtain libcdio handler");
+    return driveStatus;
+  }
+
+  CdIo_t* cdio = libCdio->cdio_open(devicePath.c_str(), DRIVER_UNKNOWN);
+  if (!cdio)
+  {
+    CLog::LogF(LOGERROR, "Failed to open device {} libcdio handler", devicePath);
+    return driveStatus;
+  }
+
+  CdioTrayStatus status = static_cast<CdioTrayStatus>(libCdio->mmc_get_tray_status(cdio));
```

Comment:
Same thing here, simply casting to an `enum class` as a 1:1 mapping isn't great. You can either:
 1) return an enum class from the `mmc_get_tray_status` method (which will likely also require mapping)
 2) add a map to translate between our enum and their int.